src/model_service.cpp

```diff
@@ -148,8 +150,21 @@ Status GetModelStatusImpl::getAllModelsStatuses(std::map<std::string, tensorflow
         }
         modelsStatusesTmp.insert({model.first, response});
     }
-    modelsStatuses.merge(modelsStatusesTmp);
 
+    const std::vector<std::string>& pipelinesNames = manager.getPipelineFactory().getPipelinesNames();
```

Comment:
Is `GetModelStatusImpl::getAllModelsStatuses` thread safe?  
Scenario where one request is calling: `manager.getPipelineFactory().getPipelinesNames();` (which is not thread safe) and model manager watcher thread is currently removing pipeline definition.